chirho/observational/handlers/soft_conditioning.py

```diff
@@ -288,12 +235,15 @@ def configure(
             torch.int32,
             torch.int64,
         ):
-            alpha = self.alpha * functools.reduce(
-                operator.mul, msg["fn"].event_shape, 1.0
-            )
-            return KernelSoftConditionReparam(
-                SoftEqKernel(alpha=alpha, event_dim=len(msg["fn"].event_shape))
-            )
+            support = constraints.boolean
+            scale = self.alpha
+
+        def _soft_eq(v1: torch.Tensor, v2: torch.Tensor) -> torch.Tensor:
+            return soft_eq(support, v1, v2, scale=scale)
+
+        # TODO decide if still needed
+        scale = scale * functools.reduce(operator.mul, msg["fn"].event_shape, 1.0)
```

Comment:
I think you can remove this rescaling step. I'm actually not even sure I should have added it in the first place, but regardless I think `AutoSoftConditioning` should no longer do anything other than set up calls to `soft_eq`. If we do want logic like that, it can live somewhere inside `soft_eq`.